app/Http/Controllers/PrintController.php

```diff
@@ -13,8 +14,23 @@ public function index() {
     }
 
     public function modify_balance(Request $request) {
+        $balance = $request->balance;
+        $validator = Validator::make($request->all(), [
+            'user_id' => 'required|integer|exists:users,id',
+            'balance' => 'required|integer',
+        ]);
+        $validator->validate();
+
+        if ($balance < 0 && User::findOrFail($request->user_id)->printAccount->balance < abs($balance)) {
+            $validator->errors()->add('balance', __('print.nobalance'));
+            return back()
+                    ->withErrors($validator)
+                    ->withInput();
+        }
+        
         User::findOrFail($request->user_id)
             ->printAccount
+            ->where('balance', '>=', abs(min($request->balance, 0)))
```

Comment:
It was a little hard to understand the logic here for me. Also incrementing negative balances (which we maybe do not use at the moment) is not possible.
I recommend to simply write the two cases separately. Also, extract the user's printAccount to a variable because it is used two times already.

Reply:
Okay.
Wouldn't we lose the atomicity if a variable would be created for the user?

Reply:
Had some debugging where it seemed like Eloquent queries don't really care about atomicity. Since it's not a major issue for us, I would go with the simplest and fastest solutions from now.